src/app/utils/utils.ts

```diff
@@ -40,25 +40,36 @@ export function getParentUrl(router: Router): string {
  * Group an array by the given property.
  *
  * @param array A simple array to be grouped.
- * @param propertyToGroupBy The key of the property in the elements by whose value the result is grouped.
+ * @param propertyToGroupBy The key of the property in the elements by whose value the result is grouped. If this is an array, grouped by individual elements.
  * @returns an array where the first entry is the value of this group and the second all entries that have this value.
  */
-export function groupBy<T, P extends keyof T>(
+export function groupBy<T, P extends keyof T, E>(
   array: T[],
   propertyToGroupBy: P,
-): [T[P], T[]][] {
+): [T[P] extends Array<E> ? E | undefined : T[P], T[]][] {
```

Comment:
```suggestion
): [T[P] extends Array<infer E> ? E | undefined : T[P], T[]][] {
```
or without `E`
```suggestion
): [T[P] extends Array<any> ? T[P][number] | undefined : T[P], T[]][] {
```
Otherwise I get `unknown` as the type for the key. With this approach I correctly get string in the test (as long as I correctly type the empty array as string too). 
See https://stackoverflow.com/a/50693866/10713841